sw/cheri/common/boot.S

```diff
@@ -7,28 +7,66 @@
 	.p2align 2
     .type start,@function
 start:
-	// ca0 (first argument) contains the read-write root
-	cspecialr        ca0, mtdc
+	// Upon reset mtdc contains the read-write root and pcc contains the execution root.
+	cspecialr        cs0, mtdc
+	auipcc           cs1, 0
 
 	// No bounds on stack, grows down from the end of IRAM
-	li               sp, 0x0010E000
-	csetaddr         csp, ca0, sp
-	auipcc           cra, 0
-	// Call the C++ entry point
-	la_abs           t0, entry_point
-	csetaddr         cra, cra, t0
-	// Store capability to stack
-	csc              cra, 0(csp)
-	// Load capability from stack
-	clc              cra, 0(csp)
-	// Clear capability from stack
-	csc              cnull, 0(csp)
```

Comment:
Removing these also removes them from CHERI sanity test. We should keep it that test and remove from all others.

Reply:
Added it to CHERI sanity via inline assembly. PTAL